File: DFS_BFS.cpp
#include<iostream>

using namespace std;

int n;
int map[30][30];
int DFSvisit[30];
int BFSvisit[30];
int rear, front;
int queue[30];

void DFS(int v) {
	DFSvisit[v]=1;

	cout << v << " ";
	for (int i = 1; i <= n; i++) {
		if (map[v][i] == 1 && !DFSvisit[i])
			DFS(i);
	}
}

void BFS(int v) {
	BFSvisit[v] = 1;
	cout << v << " ";
	queue[rear++] = v;
	while (front < rear) {
		v = queue[front++];
		for (int i = 1; i <= n; i++) {
			if (map[v][i] == 1 && !BFSvisit[i]) {
				BFSvisit[i] = 1;
				cout << i << " ";
				queue[rear++] = i;
			}
		}
	}
}

int main() {
	int start, m;
	int v1, v2;

	cin >> n >> m >> start;

	for (int i = 0; i < m; i++) {
		cin >> v1 >> v2;
		map[v1][v2] = map[v2][v1] = 1;
	}
	DFS(start);
	cout << endl;
	BFS(start);

	cin >> start;

	return 0;
}

File: DFS_BFS.py
import sys

numCases = sys.stdin.readline().rstrip().split(' ')

for i in range(3):
    numCases[i] = int(numCases[i])

arr = [[0]*(numCases[0]+1) for i in range(numCases[0]+1)]

for i in range(1, numCases[1]+1):
    num = sys.stdin.readline().rstrip().split(' ')
    num[0] = int(num[0])
    num[1] = int(num[1])
    arr[num[0]][num[1]] = 1
    arr[num[1]][num[0]] = 1
    
visited = [0] *( numCases[0]+1)


class Queue:
    def __init__(self, capacity):
        self.capacity = capacity
        self.front = -1
        self.rear = -1
        self.arr = [0] * capacity

    def push(self, num):
        self.rear+=1
        self.arr[self.rear] = num
    
    def pop(self):
        if self.front == self.rear:
            return -1
        else:
            self.front += 1
            return self.arr[self.front]

    def size(self):
        return self.rear - self.front
    
    def isEmpty(self):
        if self.rear == self.front :
            return 1
        else :
            return 0
    
    def arrfront(self):
        if self.rear == self.front:
            return -1
        else :
            return self.arr[(self.front)+1]

    def back(self):
        if self.rear == self.front:
            return -1
        else :
            return self.arr[(self.rear)]

    
def DFS(startpoint):
    visited[startpoint] = 1

    print(startpoint, end=" ")

    for i in range(1, numCases[0]+1):
        if visited[i] == 0 and arr[startpoint][i] ==1:
            DFS(i)

def BFS(startpoint):
    visited[startpoint] = 1
    print(startpoint,end = " ")
    queue = Queue(numCases[0]+1)
    queue.push(startpoint)

    while(queue.isEmpty() == 0):
        tmp = queue.pop()

        for i in range(1,numCases[0]+1):
            if visited[i] == 0 and arr[tmp][i] ==1 :
                queue.push(i)
                visited[i] = 1
                print(i, end=" ")


def main():

    DFS(numCases[2])
    print()

    for i in range(0, numCases[0]+1):
        visited[i] = 0

    BFS(numCases[2])
    
if __name__ == "__main__":
    main()